testing/trino-testing/src/main/java/io/trino/testing/AbstractTestQueryFramework.java

```diff
@@ -75,29 +83,84 @@
     private SqlParser sqlParser;
     private final AutoCloseableCloser afterClassCloser = AutoCloseableCloser.create();
     private io.trino.sql.query.QueryAssertions queryAssertions;
+    private Map<Handler, Level> oldLevels;
+    private MemoryHandler memoryHandler;
 
     @BeforeClass
     public void init()
             throws Exception
     {
+        enableLogBuffering();
         queryRunner = afterClassCloser.register(createQueryRunner());
         h2QueryRunner = afterClassCloser.register(new H2QueryRunner());
         sqlParser = new SqlParser();
         queryAssertions = new io.trino.sql.query.QueryAssertions(queryRunner);
     }
 
+    private void enableLogBuffering()
+    {
+        if (System.getenv("CONTINUOUS_INTEGRATION") == null) {
+            return;
+        }
+        // change the log level of the console handler
+        Logger rootLogger = Logger.getLogger("");
+        Handler[] handlers = rootLogger.getHandlers();
+        oldLevels = new HashMap<>();
+        Handler consoleHandler = null;
+        for (Handler handler : handlers) {
+            if (!(handler instanceof ConsoleHandler)) {
+                continue;
+            }
+            oldLevels.put(handler, handler.getLevel());
+            handler.setLevel(Level.OFF);
```

Comment:
Logging is global, and tests execute in parallel.
- Concurrently to a test extending from ATQF, there might be some other test that doesn't, and we switch its logs off
- when two instances of ATQF initialize concurrently, the second one may record `oldLevels` as `OFF`s, and dutifully restore them. This would disable logging for ever, for all tests

Reply:
What controls this? I naively thought parallelism is limited to classes: https://github.com/trinodb/trino/blob/master/pom.xml#L79

Reply:
`methods` probably means that any two methods can run in parallel, so including two methods for same class, for same test instance, and for different classes and instances
`classes` probably means that methods from a single class are not supposed to run in parallel (not sure how this plays with class hierarchies)

Reply:
I found [this article](https://howtodoinjava.com/testng/testng-executing-parallel-tests/) that has some examples with multiple classes and looks like you're right. I'll close this PR then.

Reply:
if i weren't right at least to some extent, i wouldn't spend time on https://github.com/apache/maven-surefire/pull/407 :)